src/main/java/de/dennisguse/opentracks/util/PreferencesUtils.java

```diff
@@ -318,4 +345,76 @@ public static void setDefaultExportDirectoryUri(SharedPreferences sharedPreferen
     public static boolean isDefaultExportDirectoryUri(SharedPreferences sharedPreferences, Context context) {
         return getDefaultExportDirectoryUri(sharedPreferences, context) != null;
     }
+
+    public static int getLayoutColumns(SharedPreferences sharedPreferences, Context context) {
+        return getInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, context.getResources().getInteger(R.integer.stats_custom_layout_columns_default));
+    }
+
+    public static void setLayoutColumns(SharedPreferences sharedPreferences, Context context, int columns) {
+        setInt(sharedPreferences, context, R.string.stats_custom_layout_columns_key, columns);
+    }
+
+    private static List<TypedArray> getMultiTypedArray(Context context, String key) {
+        List<TypedArray> typedArrays = new ArrayList<>();
+
+        try {
+            Class<R.array> resource = R.array.class;
+            Field field;
+            int counter = 0;
+
+            do {
+                field = resource.getField(key + "_" + counter);
+                typedArrays.add(context.getResources().obtainTypedArray(field.getInt(null)));
+                counter++;
+            } while (field != null);
```

Comment:
always true according to AndroidStudio

Reply:
when getField will fail because it doesn't exists then field will be null.